File: packages/shared/src/types/category.ts
/**
 * 分类相关类型 — 镜像后端 CategoryNode
 */

export interface CategoryNode {
  id: string;
  name: string;
  slug: string;
  iconUrl: string | null;
  sortOrder: number;
  isActive: boolean;
  children: CategoryNode[];
}

File: apps/android/app/src/main/java/com/example/shop/feature/menu/data/model/Category.kt
package com.example.shop.feature.menu.data.model

import kotlinx.serialization.Serializable

@Serializable
data class CategoryNode(
    val id: String,
    val name: String,
    val slug: String,
    val iconUrl: String? = null,
    val sortOrder: Int,
    val isActive: Boolean,
    val children: List<CategoryNode> = emptyList(),
)
